.github/workflows/code-quality.yaml

```
@@ -153,8 +184,12 @@ jobs:
     runs-on: ubuntu-26.04
     permissions:
       contents: read
+      code-quality: write  # upload coverage to GitHub Code Quality
```

actionlint (1.7.12)

[error] unknown permission scope "code-quality". all available permission scopes are "actions", "artifact-metadata", "attestations", "checks", "contents", "deployments", "discussions", "id-token", "issues", "models", "packages", "pages", "pull-requests", "repository-projects", "security-events", "statuses"

(permissions)